File: LeetCode/[M] - 199. Binary Tree Right Side View/solution.go
package main

// TreeNode - Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func rightSideView(root *TreeNode) []int {
	if root == nil {
		return []int{}
	}
	res := make([]int, 0, 100)
	queue := make([]*TreeNode, 0, 100)
	queue = append(queue, root)
	for len(queue) != 0 {
		res = append(res, queue[len(queue)-1].Val)
		lenQ := len(queue)
		for i := 1; i <= lenQ; i++ {
			v := queue[0]
			queue = queue[1:]
			if v.Left != nil {
				queue = append(queue, v.Left)
			}
			if v.Right != nil {
				queue = append(queue, v.Right)
			}
		}
	}
	return res
}

File: LeetCode/[M] - 199. Binary Tree Right Side View/solution.py
import collections
from typing import List, Optional


# TreeNode - Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        res = []
        queue = collections.deque([root])
        len_q = 1
        while queue:
            res.append(queue[-1].val)
            for _ in range(len_q):
                v = queue.popleft()
                len_q -= 1
                if v.left:
                    queue.append(v.left)
                    len_q += 1
                if v.right:
                    queue.append(v.right)
                    len_q += 1
        return res
